src/test/java/com/google/cloud/genomics/dataflow/pipelines/CountReadsITCase.java

```diff
@@ -44,22 +53,63 @@ public void voidEnsureEnvVar() {
   public void testLocal() throws Exception {
     final String OUTPUT = TEST_OUTPUT_GCS_FOLDER + "CountReadsITCase-testLocal-output.txt";
     String[] ARGS = {
+        "--apiKey=" + API_KEY,
+        "--project=genomics-pipelines",
+        "--output=" + OUTPUT,
+        "--references=1:550000:560000",
+        "--BAMFilePath=" + TEST_BAM_FNAME
+    };
+    final long EXPECTED = 685;
+    GenomicsOptions popts = PipelineOptionsFactory.create().as(GenomicsOptions.class);
+    popts.setApiKey(API_KEY);
+    GcsUtil gcsUtil = new GcsUtil.GcsUtilFactory().create(popts);
+    touchOutput(gcsUtil, OUTPUT);
+
+    CountReads.main(ARGS);
+
+    BufferedReader reader = new BufferedReader(Channels.newReader(gcsUtil.open(GcsPath.fromUri(OUTPUT)), "UTF-8"));
+    long got = Long.parseLong(reader.readLine());
+
+    Assert.assertEquals(EXPECTED, got);
+  }
+
+  /**
+   * CountReads running on Dataflow.
+   */
+  @Test
+  public void testCloud() throws Exception {
+    final String OUTPUT = TEST_OUTPUT_GCS_FOLDER + "CountReadsITCase-testCloud-output.txt";
+    String[] ARGS = {
         "--apiKey="+API_KEY,
         "--project=genomics-pipelines",
         "--output=" + OUTPUT,
+        "--numWorkers=2",
+        "--runner=BlockingDataflowPipelineRunner",
+        "--stagingLocation=" + TEST_STAGING_GCS_FOLDER,
         "--references=1:550000:560000",
-        "--BAMFilePath="+TEST_BAM_FNAME
+        "--BAMFilePath=" + TEST_BAM_FNAME
     };
     final long EXPECTED = 685;
     GenomicsOptions popts = PipelineOptionsFactory.create().as(GenomicsOptions.class);
     popts.setApiKey(API_KEY);
     GcsUtil gcsUtil = new GcsUtil.GcsUtilFactory().create(popts);
+    touchOutput(gcsUtil, OUTPUT);
 
     CountReads.main(ARGS);
 
     BufferedReader reader = new BufferedReader(Channels.newReader(gcsUtil.open(GcsPath.fromUri(OUTPUT)), "UTF-8"));
     long got = Long.parseLong(reader.readLine());
+
     Assert.assertEquals(EXPECTED, got);
   }
 
+  /**
+   * make sure we can get to the output, and at the same time avoid a false negative if
```

Comment:
You probably already figured this out, but since your importing the following:

```
import com.google.api.services.storage.Storage;
```

You can utilize the following classes to remove either the buckets, or specific objects inside them:

https://developers.google.com/resources/api-libraries/documentation/storage/v1/java/latest/com/google/api/services/storage/Storage.Buckets.Delete.html

https://developers.google.com/resources/api-libraries/documentation/storage/v1/java/latest/com/google/api/services/storage/Storage.Objects.Delete.html

It basically is one of these:

```
storage.buckets().delete("mybucket").execute();
```

or

```
storage.objects().delete("mybucket", "myobject").execute();
```

Below are the two links that refer to these:

https://cloud.google.com/storage/docs/json_api/v1/buckets/delete#examples

https://cloud.google.com/storage/docs/json_api/v1/objects/delete#examples


Hope it helps,
~p